client/state/jetpack-connect/reducer.js

```diff
@@ -84,7 +89,7 @@ export function jetpackConnectAuthorize( state = {}, action ) {
 			return Object.assign( {}, state, { siteReceived: true, isAuthorizing: false } );
 		case JETPACK_CONNECT_QUERY_SET:
 			const queryObject = Object.assign( {}, action.queryObject );
-			return Object.assign( {}, defaultAuthorizeState, { queryObject: queryObject } );
+			return Object.assign( {}, state, defaultAuthorizeState, { queryObject: queryObject } );
```

Comment:
I think `state` should be after `defaultAuthorizeState` so it rewrites the default values and not the opposite.